invoices.go

```diff
@@ -0,0 +1,186 @@
+package godo
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const invoicesBasePath = "v2/customers/my/invoices"
+
+// InvoicesService is an interface for interfacing with the Invoice
+// endpoints of the DigitalOcean API
+// See: https://developers.digitalocean.com/documentation/v2/#invoices
+type InvoicesService interface {
+	Get(context.Context, string, *ListOptions) (*Invoice, *Response, error)
+	List(context.Context, *ListOptions) (*InvoiceList, *Response, error)
+	GetSummary(context.Context, string) (*InvoiceSummary, *Response, error)
+}
+
+// InvoicesServiceOp handles communication with the Invoice related methods of
+// the DigitalOcean API.
+type InvoicesServiceOp struct {
+	client *Client
+}
+
+var _ InvoicesService = &InvoicesServiceOp{}
+
+// Invoice represents a DigitalOcean Invoice
+type Invoice struct {
+	InvoiceItems []InvoiceItem `json:"invoice_items"`
+	Links        *Links        `json:"links"`
+	Meta         *Meta         `json:"meta"`
+}
+
+// InvoiceItem represents a line-item on a DigitalOcean Invoice
+type InvoiceItem struct {
+	Product          string    `json:"product"`
+	ResourceID       string    `json:"resource_id"`
+	ResourceUUID     string    `json:"resource_uuid"`
+	GroupDescription string    `json:"group_description"`
+	Description      string    `json:"description"`
+	Amount           string    `json:"amount"`
+	Duration         string    `json:"duration"`
+	DurationUnit     string    `json:"duration_unit"`
+	StartTime        time.Time `json:"start_time"`
+	EndTime          time.Time `json:"end_time"`
+	ProjectName      string    `json:"project_name"`
+}
+
+// InvoiceList contains a paginated list of all of a customer's invoices.
+// The InvoicePreview is the month-to-date usage generated by DigitalOcean.
+type InvoiceList struct {
+	Invoices       []InvoiceListItem `json:"invoices"`
+	InvoicePreview InvoiceListItem   `json:"invoice_preview"`
+	Links          *Links            `json:"links"`
+	Meta           *Meta             `json:"meta"`
+}
+
+// InvoiceListItem contains a small list of information about a customer's invoice.
+// More information can be found in the Invoice or InvoiceSummary
+type InvoiceListItem struct {
```

Comment:
> contains a small list of information about a customer's invoice

~is it info about an invoice, or info about an invoice line item?~

Reply:
Maybe we can call this `InvoiceInfo` or `InvoiceInfoList` instead for more clarity?

Reply:
I struggled a bit with this name, it's a small set of information to identify the invoice, but less than the summary, so really an `InvoiceList` result item.  I'm not sure if either of those names are more clear about that?  This is info about a the invoice itself

Reply:
Hmm. Maybe `CustomerInvoiceInfo`? Idk, `InvoiceListItem` doesn't seem obvious to me. Mainly the list part. `InvoiceList...` makes it sound like I should be expecting a list of invoices. But yeah, I agree the naming is tough. If it makes sense to you to leave it how it is, I'm ok with it. The comment is descriptive enough too.

Reply:
Let's stick with what you have for consistency with our other code.